packages/terra-paginator/src/ProgressivePaginator.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import IconNext from 'terra-icon/lib/icon/IconNext';
+import IconPrevious from 'terra-icon/lib/icon/IconPrevious';
+import NumberField from 'terra-form/lib/NumberField';
+import ResponsiveElement from 'terra-responsive-element';
+
+import 'terra-base/lib/baseStyles';
+import styles from './Paginator.scss';
+
+import { calculatePages } from './_paginationUtils';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Function to be executed when a navigation element is selected.
+   */
+  onPageChange: PropTypes.func.isRequired,
+  /**
+   * The active/selected page.
+   */
+  selectedPage: PropTypes.number.isRequired,
+  /**
+   * Total number of all items being paginated.
+   */
+  totalCount: PropTypes.number.isRequired,
+  /**
+   * Total number of items per page.
+   */
+  itemCountPerPage: PropTypes.number.isRequired,
+};
+
+class ProgressivePaginator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { selectedPage, totalCount } = this.props;
+
+    this.state = {
+      pageInput: selectedPage,
+      selectedPage: selectedPage && totalCount ? selectedPage : undefined,
+    };
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+    this.handlePageInput = this.handlePageInput.bind(this);
+    this.buildPageSelectorForm = this.buildPageSelectorForm.bind(this);
+    this.defaultProgressivePaginator = this.defaultProgressivePaginator.bind(this);
+    this.reducedProgressivePaginator = this.reducedProgressivePaginator.bind(this);
+  }
+
+  handlePageInput(e) {
+    this.setState({ pageInput: e.target.value });
+  }
+
+  handlePageChange(index) {
+    return (event) => {
+      event.preventDefault();
+
+      this.props.onPageChange(index);
+      this.setState({ pageInput: index, selectedPage: index });
+    };
+  }
+
+  buildPageSelectorForm(totalPages, onSubmit) {
+    const { pageInput } = this.state;
+    const inputAttrs = { className: cx('page-input') };
+
+    const pageForm = (<form className={cx('page-form')} onSubmit={onSubmit(parseInt(pageInput, 10))}>
+      Page {<NumberField inputAttrs={inputAttrs} value={pageInput} onChange={this.handlePageInput} min={1} max={totalPages} />} of {totalPages}
```

Comment:
Commented in the package.json about this, let's switch to using the `InputField` from terra-form-input instead of the `NumberField` from terra-form. The [terra-form package is set to be deprecated](https://github.com/cerner/terra-core/issues/984).

Changing `NumberField` to this should work
```
<InputField inputAttrs={inputAttrs} value={pageInput} onChange={this.handlePageInput} type="number" min={1} max={totalPages} />
```

Reply:
https://github.com/cerner/terra-core/pull/1445/commits/fd545ca9b18ee3883c109a58fe3be398ad95d18c